server/src/main/java/org/elasticsearch/action/support/SubscribableListener.java

```diff
@@ -32,13 +32,11 @@
 
 /**
  * An {@link ActionListener} to which other {@link ActionListener} instances can subscribe, such that when this listener is completed it
- * fans-out its result to the subscribed listeners.
+ * fans-out its result to the subscribed listeners. Exceptions are passed to subscribed listeners without modification.
  * <p>
- * Similar to {@link ListenableActionFuture} and {@link ListenableFuture} except for its handling of exceptions: if this listener is
- * completed exceptionally then the exception is passed to subscribed listeners without modification.
```

Comment:
Discussing child classes doesn't explain what this class does (also layer violation). The child classes themselves explain the differences well enough.

Reply:
These child classes pre-date `SubscribableListener` by many years, and did not have this common superclass until quite recently, so I think it's useful to mention them here for the sake of folk that know the children but not the parent.

Reply:
Returned the references